dashboard/db/migrate/20190320200540_add_school_year_to_ap_school_codes.rb

```diff
@@ -0,0 +1,25 @@
+class AddSchoolYearToApSchoolCodes < ActiveRecord::Migration[5.0]
+  def up
+    add_column :ap_school_codes, :school_year, :integer, first: true
+
+    # Need to remove foreign key before index can be removed
+    # Readding foreign key once index is removed
+    remove_foreign_key :ap_school_codes, :schools
+    remove_index :ap_school_codes, :school_id
+    add_foreign_key :ap_school_codes, :schools
+
+    # New index created that is a composite
+    # of school code and school year
+    remove_index :ap_school_codes, :school_code
+    add_index :ap_school_codes, [:school_code, :school_year], unique: true
+  end
+
+  def down
+    remove_index :ap_school_codes, column: [:school_code, :school_year]
+
+    remove_column :ap_school_codes, :school_year
```

Comment:
Worried about this line -- if I add new rows so there are multiple values per school (ie, one per school year -- the goal of this migration), removing this column in a rollback will fail since that is one of the keys?

Reply:
do we have a policy that all our migrations be revertible? It seems like you would have to first remove the duplicate rows before running the rollback. @Hamms  might know more about how we typically handle such cases.

Reply:
@hamms any thoughts here?

Reply:
oh, sorry I missed this last week!

The primary goal of the `down` migration as I understand it is to make sure we can easily undo the migration if something goes wrong with or around it in the short-term. It's fine if migrations in the long term become unrevertible. I think this should be fine!

Reply:
Ok cool, thanks @Hamms -- I rolled this back and forward several times in the course of setting this up, so I can confirm that it works for that purpose.